### PR TITLE
add UMAP script

### LINKED ISSUE #569: Updated analysis: add code to generate methylation UMAP for MB SHH samples
#### What analysis module should be updated and why?

update `molecular-subtyping-MB/` analysis module to create a UMAP of MB SHH samples by DNA methylation. 

#### What input data should be used? Which data were used in the version being updated?

`data/methyl-beta-values.rds`

#### When do you expect the revised analysis will be completed?

1-2 days

#### Who will complete the updated analysis?

@rjcorb

### PR DESCRIPTION
**Pull request by rjcorb**
#### What scientific question is your analysis addressing?

Closes #569. This PR adds a script to `molecular-subtyping-MB/` to generate a UMAP of MB SHH samples based on most variable methylation array probes. 

#### What was your approach?

adapted from this code: https://jharenza.github.io/mb-subtypes/explore-medulloblastoma-unsupervised-clustering.nb.html

#### What GitHub issue does your pull request address?

#569 

#### Which areas should receive a particularly close look?

Check code and ensure script runs successfully: 

`R -e "rmarkdown::render('06-mb-shh-umap.Rmd')"`

#### Is there anything that you want to discuss further?

I have the UMAP points sized by age range currently, to add some additional info on outliers and un-subtyped MB SHH samples. We may not want to keep this in the final figure. 

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes

#### What types of results are included (e.g., table, figure)?

UMAP

#### What is your summary of the results?

Overall we see good clustering of MB SHH samples into the four subtypes. There are a few samples not cluttering with their subtype, including a few adolescent/adult samples with the infantile MB SHH samples. We may need to review these samples to check how they were classified into subtypes. 

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

**Comment by jharenza** [comment]
hi @rjcorb can you stack the MB PRs on https://github.com/d3b-center/OpenPedCan-analysis/pull/571? I had to remove MM2S code there.

File: analyses/molecular-subtyping-MB/06-mb-shh-umap.Rmd [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+  labs(color = "molecular subtype") +
+  theme_bw() +
+  xlab("UMAP1") +
+  ylab("UMAP2") 
+
+ggsave(file.path(plots_dir, "umap_mb.pdf"),
+       width = 5.5, height = 3.5)
+```
+
+Identify 10k most variable probes among MB Group 3/4 samples
```
Comment (zzgeng) [inline analyses/molecular-subtyping-MB/06-mb-shh-umap.Rmd]: This night be a typo, do you mean 20k?

**Comment by jharenza** [comment]
@rjcorb can you do me a favor and label the following samples on the UMAP? These have very high-conf non-MB histologies and I wonder if these are some of the problematic samples.
7316-1884
7316-1676
7316-1666
7316-3202

**Review by jharenza — COMMENTED** [review]
Can you also create an mb umap by the 4 subtypes with the expression data?

**Pull request closed without merging**